vtex-apps/product-customizer
# Add recursive dispatch update

#### What problem is this solving?
Fix the `assemblyOption` structure that is passed to addToCart query. 

#### How to test it?
[Workspace](https://brasileirovtex--acctglobal.myvtex.com/mcproductrefid136/p)

**Test 1**
1. Go into workspace
2. Click to customize combo 1
3. Select `Lanche Genérico`
4. Repeat steps 2 and 3 to the other assembly option
5. Click to add product to cart
6. Look at minicart if `Lanche Generico` appears

**Test 2**
1. Go into workspace
2. Click to customize combo 1
3. Select `Lanche Genérico`
4. Repeat steps 2 and 3 to the other assembly option
5. Click to add product to cart
6. Look at minicart if `Lanche Generico` appears
7. Customize `Lanche Generico` adding some extras values


#### Screenshots or example usage:

**After**:
![image](https://user-images.githubusercontent.com/15680320/106316337-4edb5b80-624b-11eb-9b03-8e0237c466ce.png)

**Before**: Pay attention that you've chosen `Lanche Genérico` not the `Big MActt`
![image](https://user-images.githubusercontent.com/15680320/106316465-8a762580-624b-11eb-9571-9fdc9b9fed06.png)


#### Describe alternatives you've considered, if any.
Change the way that add-cart-to-button create the variables to addToCart mutation, look this PR https://github.com/vtex-apps/add-to-cart-button/pull/61

#### Related to / Depends on
-

#### How does this PR make you feel? [:link:](http://giphy.com/)
![](https://media.giphy.com/media/L0Z4qwdwv62cn4haFp/giphy.gif)
